packages/thumbprint-react/components/Popover/index.jsx

```diff
@@ -0,0 +1,219 @@
+
+    const handleKeyUp = event => {
+        if (event.keyCode === ESC_KEY) {
+            onCloseClick();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('keyup', handleKeyUp);
+        return () => {
+            document.removeEventListener('keyup', handleKeyUp);
+        };
+    });
+
+    Popover.handleClickOutside = () => onCloseClick();
+
+    const popoverContent = (
+        <PopoverContent position={position} isOpen={isOpen} onCloseClick={onCloseClick}>
+            {children}
+        </PopoverContent>
+    );
+
+    return (
+        <Manager>
+            <Reference>{({ ref }) => launcher({ ref })}</Reference>
+            {canUseDOM &&
+                (shouldDisplace
+                    ? ReactDOM.createPortal(popoverContent, document.body)
+                    : popoverContent)}
+        </Manager>
+    );
+};
+
+Popover.propTypes = {
+    /**
+     * Contents for the Popover. Usually a `PopoverTitle`, `PopoverBody`, and `PopoverPrimaryButton`
+     */
+    children: PropTypes.node,
+    /**
+     * A function that renders JSX and receives an object with `ref`.
+     * All of these props must be added to the component within the render prop.
+     */
+    launcher: PropTypes.func.isRequired,
+    /**
+     * Position of popover relative to the launcher.
+     */
+    position: PropTypes.oneOf([
+        'top-start',
+        'top',
+        'top-end',
+        'bottom-start',
+        'bottom',
+        'bottom-end',
+        'left-start',
+        'left',
+        'left-end',
+        'right-start',
+        'right',
+        'right-end',
+    ]),
+    /**
+     * Whether or not the popover is visible.
+     */
+    isOpen: PropTypes.bool.isRequired,
+    /**
+     * Function called when the close button is clicked. You should cause this to set `isOpen=false`
+     * in your parent component
+     */
+    onCloseClick: PropTypes.func.isRequired,
+    /**
+     * By default popovers will render right before the `</body>` tag.
+     * Setting the `container` to `inline` causes the tooltip to remain where it was added to the
+     * DOM.
+     * This option is helpful to work around z-index and positioning issues.
+     */
+    container: PropTypes.oneOf(['inline', 'body']),
+};
+
+Popover.defaultProps = {
+    children: null,
+    position: 'top',
+    container: 'body',
+};
+
+const clickOutsideConfig = {
+    handleClickOutside: () => Popover.handleClickOutside,
+};
+
+export default onClickOutside(Popover, clickOutsideConfig);
+
+const PopoverTitle = ({ children }) => <div className={styles.popoverTitle}>{children}</div>;
+
+PopoverTitle.propTypes = {
+    /**
+     * A title for the popover.
+     */
+    children: PropTypes.string.isRequired,
+};
+
+const PopoverBody = ({ children }) => <div className={styles.popoverBody}>{children}</div>;
+
+PopoverBody.propTypes = {
+    /**
+     * The body description text for the popover.
+     */
+    children: PropTypes.string.isRequired,
```

Comment:
It's possible that someone would include a link within the text. Should it be a `node` instead?